source/grammar/openpulseParser.g4

```diff
@@ -0,0 +1,70 @@
+/***** ANTLRv4  grammar for OpenPulse. *****/
+// This grammar is OpenPulse extension for QOpenQASM3.0
+// We introduce several new types for OpenPulse.
+// We override several nodes in qasm3 to consume the new types.
+// The OpenPulse grammar can be used inside cal or defcal blocks.
+
+parser grammar openpulseParser;
+import qasm3Parser;
```

Comment:
We import the qasm3Parser, override `defcalStatement` with concrete implementation of OpenPulse. We also override `scalarType` to incorporate `OpenPulse` types and override `statement` to add `calStatement`.

Reply:
My interpretation of the spec is that the calibration grammar components don't take effect until the _block_ of the `defcal` is entered, so parsing the identifiers should be the responsibility of the OpenQASM 3 component of the parser.  This particular case might be a bug in the OQ 3 ANTLR grammar - I have a small comment on the bottom of #367 about it, but assuming we want to keep that, I'll update the OQ 3 grammar.

Reply:
> My interpretation of the spec is that the calibration grammar components don't take effect until the block of the defcal is entered, so parsing the identifiers should be the responsibility of the OpenQASM 3 component of the parser

That is true. However, the `defcal` body can contain classical statements. We took the liberty of reusing the OpenQASM classical statements in OpenPulse (so we don't have to introduce a different grammar). Therefore we try to reuse qasm3Parser to pase the classical statements in OpenPulse.